### ISSUE TITLE
[stacked] [stacked_generator] [stacked_services] Can't navigate back with data

### ISSUE THREAD

**Issue by kodjosama** [Stacked-Org/stacked#498]
Hi, I'm trying to navigate back with some data from a view model but got an error.

```dart
======== Exception caught by gesture ===============================================================
The following _TypeError was thrown while handling a gesture:
type '_InternalLinkedHashMap<String, bool>' is not a subtype of type 'MaterialRoute<dynamic>?' of 'result'

When the exception was thrown, this was the stack: 
#0      LocalHistoryRoute.didPop (package:flutter/src/widgets/routes.dart)
#1      _RouteEntry.pop (package:flutter/src/widgets/navigator.dart:3085:15)
#2      NavigatorState.pop (package:flutter/src/widgets/navigator.dart:5077:13)
#3      GetNavigation.back (package:get/get_navigation/src/extension_navigation.dart:860:34)
#4      NavigationService.back (package:stacked_services/src/navigation_service.dart:172:9)
...
====================================================================================================
```

**Description**
I have two views, _RegisterView_ and _ForgotPassword_, and they each call _CheckOtpCodeView_ to verify if the user owns the phone number.

**Here is the code**

_app.dart_

```dart
@StackedApp(
  routes: const [
    MaterialRoute(page: RegisterView, initial: true),
    MaterialRoute(page: CheckOtpCodeView),
  ],
  dependencies: [
    // third app services
    LazySingleton(classType: DialogService),
    LazySingleton(classType: NavigationService),

    // app services
  ],
)
class App {}
```

_RegisterViewModel_

```dart
// 

Future<void> verifyPhoneNumber() async {
    final response = await navigationService.navigateTo(
      Routes.checkOtpCodeView,
      arguments: CheckOtpCodeViewArguments(
        phoneNumber: phoneNumberValue!,
      ),
    );

    print('response $response');
  }
```

_CheckOtpCodeViewModel_

```dart
void markAsVerified() {
    navigationService.back(result: {'verified': true});
  }

  void markAsUnverified() {
    navigationService.back(result: {'verified': false});
  }
```

But when I'm doing this, didn't throw exceptions

```dart
void markAsVerified() {
    navigationService.back(result: MaterialRoute(page: CheckOtpCodeView)); // !!??
  }
```

**Comment by danilof** [comment]
Hi I have the same issue here.

It looks like router generator in app.router.dart generate wrong return type.

When I change from:
`return MaterialPageRoute<MaterialRoute<dynamic>>`
to exact type:
`return MaterialPageRoute<List<CustomerOrderDetail>>`
it start working again.

Is there another way to solve this?

**Comment by josephmangmang** [comment]
Same issue here

**Comment by josephmangmang** [comment]
I created a pr
https://github.com/FilledStacks/stacked/pull/509

**Comment by FilledStacks** [comment]
PR merged. Will update version and make a new release of `stacked_generator`

**Comment by FilledStacks** [comment]
Released in generator 0.5.3

**Comment by sqrg** [comment]
I'm having issues with this. How do you define a route for the generator to correctly use the return type?
